src/main/java/com/mojang/brigadier/StringReader.java

```diff
@@ -239,4 +239,21 @@ public void expect(final char c) throws CommandSyntaxException {
         }
         skip();
     }
+
+    public void expect(final String s) throws CommandSyntaxException {
+        if(!canRead(s.length()) || !getRemaining().startsWith(s)) {
+            throw CommandSyntaxException.BUILT_IN_EXCEPTIONS.readerExpectedSymbol().createWithContext(this, s);
+        }
+        this.cursor += s.length();
+    }
+
+    public String expectOption(final String ...opts) throws CommandSyntaxException {
```

Comment:
I'm a little unsure about this. If you did `expectOption("foo", "bar", "footprint", "barstool")` on the string `"footprint"` you'd get `"foo"` as a return. Technically correct, but would that be clear? It's also not nearly as efficient as just taking the string and comparing afterwards.

Can you provide me some use cases of how this is intended to be used? I can only imagine trying to parse literals, but that's handled in another way.

Reply:
Style issue: do `final String... opts`

(I'm sorry - this is my fault for not commiting a style guide yet. I'll do that as soon as I can.)

Reply:
OK. I guess it makes sense for expectOption to match the longest instead of the first

Reply:
It would be used for things like matching `}` or `,`.